graphsdk/src/androidTest/java/com/microsoft/graph/concurrency/DefaultExecutorsTests.java

```diff
@@ -41,35 +40,26 @@ public void run() {
         });
         // sleep 1 second for background task
         SystemClock.sleep(1000);
+        new SimpleWaiter().signal();
+        assertFalse(Looper.getMainLooper().isCurrentThread());
         assertTrue(success.get());
         assertEquals(1,mLogger.getLogMessages().size());
-        assertTrue(mLogger.isExist(expectedLogMessage));
+        assertTrue(mLogger.hasMessage(expectedLogMessage));
     }
 
     public void testPerformOnForegroundWithResult() {
         String expectedResult = "result value";
         String expectedLogMessage = "Starting foreground task, current active count:0, with result result value";
-        final AtomicBoolean success = new AtomicBoolean(false);
-        final AtomicBoolean failure = new AtomicBoolean(false);
-        ICallback<String> callback = new ICallback<String>() {
-            @Override
-            public void success(String s) {
-                success.set(true);
-                callbackResult = s;
-            }
-
-            @Override
-            public void failure(ClientException ex) {
-                failure.set(true);
-            }
-        };
+        MockCallback callback = new MockCallback();
         defaultExecutors.performOnForeground(expectedResult,callback);
         SystemClock.sleep(1000);
-        assertTrue(success.get());
-        assertFalse(failure.get());
-        assertEquals(expectedResult, callbackResult);
+        new SimpleWaiter().signal();
+        assertFalse(Looper.getMainLooper().isCurrentThread());
+        assertTrue(callback.getSuccess());
+        assertFalse(callback.getFailure());
+        assertEquals(expectedResult, callback.getResult());
         assertEquals(1,mLogger.getLogMessages().size());
-        assertTrue(mLogger.isExist(expectedLogMessage));
+        assertTrue(mLogger.hasMessage(expectedLogMessage));
     }
 
     public void testPerformOnForegroundWithProgress() {
```

Comment:
Please review the following code for this test case, its is a much cleaner example that can be re-used in all of the test cases.

```java
    public void testProgressIsCalledOnForeground() throws Exception {
        // Setup
        final String expectedLogMessage = "Starting foreground task, current active count:0, with progress  1, max progress1";
        final ExecutorTestCallback<String> callback = new ExecutorTestCallback<>();
        final long expectedCurrentValue = 1234;
        final long expectedMaxValue = 4321;

        // Action
        defaultExecutors.performOnForeground(expectedCurrentValue, expectedMaxValue, callback);
        
        // Verification
        callback._completionWaiter.waitForSignal();
        Assert.assertEquals(CallingState.Foreground, callback._callingState.get());
        Assert.assertEquals(false, callback._successCalled.get());
        Assert.assertEquals(false, callback._failureCalled.get());
        Assert.assertEquals(true, callback._progressCalled.get());
        Assert.assertEquals(expectedCurrentValue, callback._progressResultCurrent.get());
        Assert.assertEquals(expectedMaxValue, callback._progressResultMax.get());
    }

    private class ExecutorTestCallback<T> implements IProgressCallback<T> {
        SimpleWaiter _completionWaiter = new SimpleWaiter();
        AtomicReference<CallingState> _callingState = new AtomicReference<>(CallingState.Unknown);

        AtomicBoolean _successCalled = new AtomicBoolean(false);
        AtomicReference<T> _successResult = new AtomicReference<>();

        AtomicBoolean _failureCalled = new AtomicBoolean(false);
        AtomicReference<ClientException> _exceptionResult = new AtomicReference<>();

        AtomicInteger _progressCalled = new AtomicInteger(0);
        AtomicLong _progressResultCurrent = new AtomicLong(-1);
        AtomicLong _progressResultMax = new AtomicLong(-1);

        @Override
        public void success(final T result) {
            _successCalled.set(true);
            _successResult.set(result);
            _callingState.set(getCallingState());
            _completionWaiter.signal();
        }

        @Override
        public void failure(final ClientException ex) {
            _failureCalled.set(true);
            _exceptionResult.set(ex);
            _callingState.set(getCallingState());
            _completionWaiter.signal();
        }

        @Override
        public void progress(final long current, final long max) {
            _progressCalled.incrementAndGet();
            _progressResultCurrent.set(current);
            _progressResultMax.set(max);
            _callingState.set(getCallingState());
            _completionWaiter.signal();
        }

        private CallingState getCallingState(){
            return Looper.getMainLooper().isCurrentThread() ? CallingState.Foreground : CallingState.Background;
        }
    }

    private enum CallingState {
        Foreground, Background, Unknown;
    }
```